.github/workflows/dev-image-update.yaml

```diff
@@ -102,6 +102,8 @@ jobs:
     needs: [ update_image, helm_tag_bump, cherry_pick_image_to_production_gcr ]
     uses: ./.github/workflows/notify-slack.yaml
     secrets: inherit
+    if: ${{ !cancelled() }}
     with:
       workflow_name: Dev Image Update
       notify_on_success: true
```

Comment:
Does this parameter still make sense? In the false case wouldn't the workflow no longer be called?

Reply:
There are three states that we can check for here - success, failure or cancelled. (Cancelled being when the job is canceled manually by a user). So, the !cancelled() check means that it will run either on success on failure. 

It's a little clunky, but it is the [suggested option in the github actions docs](https://docs.github.com/en/actions/learn-github-actions/expressions#always). 

>  If you want to run a job or step regardless of its success or failure, use the recommended alternative: if: ${{ !cancelled() }}

But, given that it's confusing, I think I'll just switch to always() and then cancelled jobs will just get the failure notification.